crd/nodenetworkconfig/client.go

```diff
@@ -136,3 +144,20 @@ func (c *Client) UpdateSpec(ctx context.Context, key types.NamespacedName, spec
 	}
 	return nnc, nil
 }
+
+// SetOwnerRef sets the owner of the NodeNetworkConfig to the given object, using HTTP Patch
+func (c *Client) SetOwnerRef(ctx context.Context, nnc *v1alpha.NodeNetworkConfig, owner metav1.Object) error {
+	newNNC := nnc.DeepCopy()
```

Comment:
I think we should either return the `newNNC` or mutate the passed `nnc`, since the patch call will write the resultant object to the ref it gets passed.

Reply:
good point, I'll change that

Reply:
decided to return instead of mutate since the `nnccli.Patch()` could fail and we don't want to mutate the caller if we dn't actually make a successful change